pumpkin-inventory/src/player.rs

```diff
@@ -57,13 +57,19 @@ impl PlayerInventory {
             if !(0..=45).contains(&slot) {
                 Err(InventoryError::InvalidSlot)?
             }
-            *self.all_slots()[slot] = item;
+            let Some(slot) = self.get_slot_mut(slot) else {
+                return Err(InventoryError::InvalidSlot);
+            };
+            *slot = item;
             return Ok(());
         }
         let slot_condition = self.slot_condition(slot)?;
         if let Some(item) = item {
             if slot_condition(&item) {
-                *self.all_slots()[slot] = Some(item);
+                let Some(slot) = self.get_slot_mut(slot) else {
```

Comment:
change this to `let slot = self.get_slot_mut(slot).ok_or(Err(InventoryError::InvalidSlot))?;`